### DISCUSSION TITLE
create an instance in a may to many implicit relationship

### DISCUSSION THREAD

**Discussion by ranran2121** [prisma/prisma#17033] [Q&A]
Hi,
i have the following (simplified) schema
```
model Profile {
  id Int @id @default(autoincrement())
  name String
  jobExpectations JobExpectation[]
  otherExpectations OtherExpectation[]
}

model JobExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile[]
}

model OtherExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile[]
}
```

I need to create an instance of the Profile table so I am using this query
```
const profile = {
   name: "Jon",
   jobExpectations:[{id:1, name:"salary"}],
   otherExpectations:[{id:1, name:"movies"}]
}

const profile = await prisma.profile.create({
      data: profile,     
    });
```
NB: jobExpectation with id=1 and name=salary already exists in its table, Same for the otherExpectation

The above query doesn't work. How am I suppose to connect the profile instance with the jobexpectation and otherexpectation instance?

**Comment by ranran2121** [comment]
@nurul3101 , could you please help me?

**Comment by knnedy** [comment]
consider using the @relation attribute
```
model Profile {
  id Int @id @default(autoincrement())
  name String
  jobExpectations JobExpectation[]
  otherExpectations OtherExpectation[]
}

model JobExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile @relation(fields: [profileId], references: [id], onDelete: Cascade)
  profileId Int
}

model OtherExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile @relation(fields: [profileId], references: [id], onDelete: Cascade)
  profileId Int
}
```
Hope it is helpful.

**Reply by ranran2121** [reply]
sorry, I do not know if that was the answer. According to the documentation [(https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations)](url), an implicit many-to many relationship can be written simply with this syntax
```
model Post {
  id         Int        @id @default(autoincrement())
  title      String
  categories Category[]
}

model Category {
  id    Int    @id @default(autoincrement())
  name  String
  posts Post[]
}
```
Now, I have a table with profile (id=1) and a table with a list of jobexpectations. This list is already created, and I needed to connect profile with id=1 with let's say jobExpectations with id=1 and id=3 and id=5. 
Eventually I solved the problem with this query:
```
const profile = {
   name: "Jon",
   jobExpectations:{
        connect: [{id:1}, {id:3}, {id:5}],
      },
   otherExpectations:{
        connect: [{id:2}, {id:4}, {id:6}],
      }
}

const profile = await prisma.profile.create({
      data: profile,     
    });
```
With the syntax you suggested, how should I have written the query to achieve the same result
(again, I do no need to create a brand new JobExpectation, since the user can select only in a certain list)

**Reply by nurul3101** [reply]
Hi @ranran2121 👋 
The query that you have used is correct and should work as expected 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
